src/uu/cksum/src/cksum.rs

```diff
@@ -358,6 +374,26 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
         None
     };
 
+    if ["bsd", "crc", "sysv"].contains(&algo_name) && check {
+        return Err(io::Error::new(
+            io::ErrorKind::InvalidInput,
+            "--check is not supported with --algorithm={bsd,sysv,crc}",
+        )
+        .into());
+    }
+
+    let untagged: bool = matches.get_flag(options::UNTAGGED);
+
+    let args: Vec<String> = std::env::args().collect();
+    let binary = if args.contains(&"--tag".to_string()) && args.contains(&"--untagged".to_string())
+    {
+        // ugly workaround to detect if the two arguments have been passed at the same time
+        // clap with overrides_with
+        false
```

Comment:
This is still unreachable, I believe.

Reply:
why do you say that?

Reply:
Ah! You're right, this *is* reachable, which is yet another bug.

Then please explain what this logic is trying to do? Because either the final outcome of parsing the command-line is "we used the tagged format" or "we use the untagged format", so it doesn't matter if both appear or not.

Oh no. I think I know what's going on: `--tag` un-sets `--binary`, so the relative position on the command-line matters. And here's a little experiment to prove it:

```console
$ ../gnu/src/cksum --binary --tag --untagged --algorithm=md5 emptyfile 
d41d8cd98f00b204e9800998ecf8427e  emptyfile # "--binary" was overwritten by "--tag"
$ ../gnu/src/cksum --tag --untagged --binary --algorithm=md5 emptyfile 
d41d8cd98f00b204e9800998ecf8427e *emptyfile
```

This means that the `if` condition is still wrong, but for a different reason. I'm afraid the only way to fix it is to either fix #4254 or use `indices_of` to detect the order, as already happens all over the place [in ls](https://github.com/uutils/coreutils/blob/main/src/uu/ls/src/ls.rs#L351).

Reply:
ok, I will see if I can fix this upstream. it is too contusing